x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -58,6 +62,23 @@ export const hostMiddlewareFactory: ImmutableMiddlewareFactory<HostState> = (cor
           type: 'serverReturnedHostList',
           payload: hostResponse,
         });
+
+        getNonExistingPoliciesForHostsList(
+          coreStart.http,
+          hostResponse.hosts,
+          nonExistingPolicies(state)
+        )
+          .then((missingPolicies) => {
```

Comment:
FYI: I don't like that these `then().catch()` blocks are repeated below. Will look to improve in the future (maybe by passing along the `dispatch` method to the `getNonexistingPoliciesForHostList()` as an argument.